src/python/pants/backend/python/framework/django/dependency_inference.py

```diff
@@ -117,6 +116,77 @@ async def django_parser_script(
     )
 
 
+async def _django_app_implicit_dependencies(
+    request: InferDjangoDependencies,
+    python_setup: PythonSetup,
+    django_apps: DjangoApps,
+) -> InferredDependencies:
+    file_path = request.field_set.source.file_path
+    apps = [
+        django_app for django_app in django_apps.values() if django_app.config_file == file_path
+    ]
+    if not apps:
+        return InferredDependencies([])
+
+    app_package = apps[0].name
+
+    implicit_dependency_packages = [
+        f"{app_package}.{subpackage}" for subpackage in _implicit_dependency_packages
+    ]
+
+    resolve = request.field_set.resolve.normalized_value(python_setup)
+
+    resolved_dependencies = await Get(
+        ResolvedParsedPythonDependencies,
+        ResolvedParsedPythonDependenciesRequest(
+            request.field_set,
+            ParsedPythonDependencies(
+                ParsedPythonImports(
+                    (package, ParsedPythonImportInfo(0, False))
+                    for package in implicit_dependency_packages
+                ),
+                ParsedPythonAssetPaths(),
+            ),
+            resolve,
+        ),
+    )
+
+    spec_paths = [
+        address.spec_path
+        for result in resolved_dependencies.resolve_results.values()
+        if result.status in (ImportOwnerStatus.unambiguous, ImportOwnerStatus.disambiguated)
```

Comment:
This may not (yet) be current practice, but I'd prefer to have a method on the resolve result that does this check. Something like: `if result.is_authoritative` (bike shed name, of course).

Motivation being, that if/when we change `ImportOwnerStatus` the impact will be much more contained and clearer to reason about, as well as making it easier to read here what the condition actually is checking.

Reply:
Absolutely. Also, this is a design observation I'm making now that I've come to employ from recent experiences working on refactoring legacy code. i.e. I've got some new eyes in that regard :)